### PR TITLE
[indexer grpc] in-memory cache benchmarks.

### PR DESCRIPTION
**Pull request by larry-aptos**
## Description

* Add benchmarks for in-memory cache.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [x] Other (specify) indexer-grpc

## How Has This Been Tested?

Locally tested.

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: .github/workflows/indexer-grpc-in-memory-cache-benchmark.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install grpcurl
+        run: curl -sSL "https://github.com/fullstorydev/grpcurl/releases/download/v1.8.7/grpcurl_1.8.7_linux_x86_64.tar.gz" | sudo tar -xz -C /usr/local/bin
+      - name: Rust setup
+        uses: aptos-labs/aptos-core/.github/actions/rust-setup@main
+        with:
+            GIT_CREDENTIALS: ${{ secrets.GIT_CREDENTIALS }}
+      - name: build and run the benchmark
```
Comment (yuunlimm) [inline .github/workflows/indexer-grpc-in-memory-cache-benchmark.yaml:18]: question: how are the results of the benchmark handled after the workflow?

File: ecosystem/indexer-grpc/indexer-grpc-in-memory-cache-benchmark/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    let tasks = (0..task_count)
+        .map(|_| spawn_cache_consumption_task(cache.clone(), 20))
+        .collect::<Vec<_>>();
+    // join all the tasks.
+    let tps = futures::future::join_all(tasks)
+        .await
+        .iter()
+        .map(|r| r.as_ref().unwrap())
+        .sum::<f64>()
+        / task_count as f64;
```
Comment (yuunlimm) [inline ecosystem/indexer-grpc/indexer-grpc-in-memory-cache-benchmark/src/main.rs:147]: nit: maybe we can parametrize transaction size and others, so we can add more test scenarios if needed by reducing duplication. 

```
#[tokio::main(flavor = "multi_thread", worker_threads = 16)]
async fn main() -> anyhow::Result<()> {
    run_transaction_test(2 * 1024, 10, 20, 20_000.0).await;
    run_transaction_test(2 * 1024, 1000, 20, 100.0).await;
    run_transaction_test(1024 * 1024, 20, 5, 500.0).await;
    Ok(())

async fn run_transaction_test(transaction_size: usize, task_count: usize, duration: u64, expected_tps: f64) {
}
```

**Review by yuunlimm — APPROVED** [review]

File: ecosystem/indexer-grpc/indexer-grpc-in-memory-cache-benchmark/src/main.rs (hunk 2 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-in-memory-cache-benchmark/src/main.rs]: good call; done

File: .github/workflows/indexer-grpc-in-memory-cache-benchmark.yaml (hunk 1 above)
Comment (yuunlimm) [reply .github/workflows/indexer-grpc-in-memory-cache-benchmark.yaml:18]: sounds good!

File: docker/builder/indexer-grpc.Dockerfile [hunk 3]
```diff
@@ -17,6 +17,7 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
 COPY --link --from=indexer-builder /aptos/dist/aptos-indexer-grpc-cache-worker /usr/local/bin/aptos-indexer-grpc-cache-worker
 COPY --link --from=indexer-builder /aptos/dist/aptos-indexer-grpc-file-store /usr/local/bin/aptos-indexer-grpc-file-store
 COPY --link --from=indexer-builder /aptos/dist/aptos-indexer-grpc-data-service /usr/local/bin/aptos-indexer-grpc-data-service
+COPY --link --from=indexer-builder /aptos/dist/aptos-indexer-grpc-in-memory-cache-benchmark /usr/local/bin/aptos-indexer-grpc-in-memory-cache-benchmark
```
[the remark this answers is not in the document]
Comment (larry-aptos) [reply docker/builder/indexer-grpc.Dockerfile:20]: Ideally yes; the size is ~2mb thus didn't create a new one for it 🤦

File: fullnode.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+base:
```
Comment (just-in-chang) [inline fullnode.yaml:1]: Did u mean to add the configs + genesis blobs?

Comment (larry-aptos) [reply fullnode.yaml]: good catch!

**Pull request merged**